File: app/src/main/java/com/homework/mhafidhaziz/weatherapp/network/WeatherApi.java
package com.homework.mhafidhaziz.weatherapp.network;

import android.database.Observable;
import com.homework.mhafidhaziz.weatherapp.network.dto.WeatherDto;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by mhafidhabdulaziz on 10/02/19.
 * weather-app
 * help.aziz@gmail.com
 * Copyright 2019
 */
public interface WeatherApi {
    @GET("forecast.json")
    Observable<WeatherDto> getWeather(@Query("key") String key,
                                      @Query("q") String location,
                                      @Query("days") String days);
}

File: app/src/main/java/com/homework/mhafidhaziz/weatherapp/network/WeatherApi.kt
package com.homework.mhafidhaziz.weatherapp.network

import com.homework.mhafidhaziz.weatherapp.network.dto.WeatherDto
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by mhafidhabdulaziz on 10/02/19.
 * weather-app
 * help.aziz@gmail.com
 * Copyright 2019
 */
interface WeatherApi {
    @GET("forecast.json")
    fun getWeather(
        @Query("key") key: String,
        @Query("q") location: String,
        @Query("days") days: String
    ): Observable<WeatherDto>
}